### PR TITLE
feat: support dual stack deployment of istio

### LINKED ISSUE #44017: Istio itself should be deployed in Dual Stack mode if Dual Stack is enabled when user deploy the Istio
### Bug Description

As the Dual Stack feature is supported as experimental status feature since 1.17, however, the Istio itself is still deployed in the single stack even if the Dual Stack is enabled by user.
IMO, we should make Istio deployed in Dual Stack as well if user definitely enabled it like this.  And this issue should be part of #40394 as well.

### Version

```prose
$ ./istio-1.17.1/bin/istioctl version
client version: 1.17.1
control plane version: 1.17.1
data plane version: 1.17.1 (11 proxies)

$ kubectl version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.25.4
Kustomize Version: v4.5.7
Server Version: v1.25.6+k3s1
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by Xunzhuo**
**Please provide a description of this PR:**

Fixes: https://github.com/istio/istio/issues/44017

### CODE REVIEW

File: releasenotes/notes/44017.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# issue is a list of GitHub issues resolved in this note.
+# If issue is not in the current repo, specify its full URL instead.
+issue:
+  - 44017
+
+# releaseNotes is a markdown listing of any user facing changes. This will appear in the
+# release notes.
+releaseNotes:
+- |
+  **Added** support Istiod Dual Stack Deployment Mode.
```
Comment (hanxiaop) [inline releasenotes/notes/44017.yaml:34]:
```suggestion
  **Added** support for Istiod Dual Stack Deployment Mode.
```

**Review by hanxiaop — DISMISSED** [review]
Releasenote looks good.

**Comment by zhlsunshine** [comment]
Besides the istiod service deployment, there should be others need to be considered, such as ingress gateway, etc.

File: manifests/charts/base/templates/services.yaml [hunk 2]
```diff
@@ -11,6 +11,12 @@ metadata:
   {{- end }}
   namespace: {{ .Release.Namespace }}
 spec:
+  {{- if .Values.pilot.env.ISTIO_DUAL_STACK }}
+  ipFamilyPolicy: RequireDualStack
+  ipFamilies:
+  - IPv6
```
Comment (zhlsunshine) [inline manifests/charts/base/templates/services.yaml:17]: Besides the `ipFamilyPolicy` here, we may need to discuss on which IP family should be the first, such as [IPv4, IPv6] or [IPv6, IPv4]. There may be dependencies or decisions on which should be used.

**Comment by hobbytp** [comment]
@Xunzhuo will you update it later? thanks,

**Comment by jorhett** [comment]
@Xunzhuo @zhlsunshine I realize that more configuration is necessary, but it seems that something like this is required to be able to terminate IPv6 at the external gateway -- which is a much more common use case that IPv6 inside the mesh.

**Pull request closed without merging**